### PR TITLE
Split box and cylindrical position limits

### PR DESCRIPTION
**Pull request by lgulich**
## Summary

- Rename the axis-aligned relation to `PositionLimitsBox`.
- Add a separate `PositionLimitsCylindrical` relation for world-XY keep-in, keep-out, and annular radial limits.
- Give each relation its own loss strategy and compose box plus cylindrical limits by attaching both relations.
- Keep the exact-center lower-radius symmetry breaker, while removing cross-geometry gradient steering.

## Backward compatibility

- `PositionLimits` remains a Python alias for `PositionLimitsBox`.
- Legacy YAML `kind: position_limits` resolves to `PositionLimitsBox`.
- Existing box constructor arguments and positional `relation_loss_weight` behavior remain unchanged.

## Validation

- 154 directly applicable relation, solver, placement-validation, and graph-spec tests pass.
- All configured pre-commit hooks pass.

### CODE REVIEW

File: isaaclab_arena/relations/relations.py [hunk 1]
```diff
@@ -493,15 +493,24 @@ def __init__(
         z_min: float | None = None,
         z_max: float | None = None,
         relation_loss_weight: float = 1.0,
+        center_x: float | None = None,
+        center_y: float | None = None,
```
Comment (xyao-nv) [inline isaaclab_arena/relations/relations.py]: Could be in another MR, and open for discussions.
Now you are expecting the anchor (aka center) is specified by user, I think this radial bound could also be helpful when the center xy comes from an anchor object xy. 
Basically treatthe radial bounds as a constraint wrt the anchor object, or even robot (once we enable robot-object co-placement https://github.com/isaac-sim/IsaacLab-Arena/pull/915)
For example,
"Droid picks up apple from the sink and places it in a bowl" -> Apple shall be within 50cm(for example) from the bowl, and within 100cm(for example) from the robot fix base.

**Review by xyao-nv — APPROVED** [review]

File: isaaclab_arena/relations/relations.py (hunk 1 above)
Comment (lgulich) [reply isaaclab_arena/relations/relations.py]: Fully agree. I was hoping to keep this relation as an "absolute position constraint" and in a separate MR add a "relative position constraint" that can do box or disk constraints around other objects that are placed as part of the optimization problem.

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 2]
```diff
@@ -617,5 +620,71 @@ def compute_loss(
...
+            else:
+                total_loss = total_loss + single_boundary_linear_loss(
+                    radius, relation.radius_max, slope=self.slope, penalty_side="greater"
+                )
+
         result = relation.relation_loss_weight * total_loss
         return result.squeeze(0) if single_input else result
+
+    @staticmethod
+    def _radius_with_lower_bound_escape_direction(
```
[the remark this answers is not in the document]
Comment (lgulich) [reply isaaclab_arena/relations/relation_loss_strategies.py]: Addressed in f75befb5. Added a focused test that directly exercises the escape branch and asserts the steered radial gradient points along [-1, 0], into the feasible axial box.

File: isaaclab_arena/relations/relations.py [hunk 3]
```diff
@@ -515,6 +524,10 @@ def __init__(
         self.z_min = z_min
         self.z_max = z_max
         self.relation_loss_weight = relation_loss_weight
+        self.center_x = center_x
+        self.center_y = center_y
+        self.radius_min = radius_min
+        self.radius_max = radius_max
```
[the remark this answers is not in the document]
Comment (lgulich) [reply isaaclab_arena/relations/relations.py]: Addressed in f75befb5. PositionLimits now rejects center_x or center_y unless a radial bound is present, with parametrized coverage for either coordinate and both coordinates.

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 4]
```diff
@@ -617,5 +620,71 @@ def compute_loss(
...
+            feasibility_steered_radius = radius.detach() + directional_radius - directional_radius.detach()
+            inside_lower_bound = radius < relation.radius_min
+            return torch.where(inside_lower_bound, feasibility_steered_radius, radius)
+
+        at_center = torch.all(radial_delta == 0, dim=1)
+        epsilon = torch.finfo(radial_delta.dtype).eps
+        positive_x_epsilon = torch.zeros_like(radial_delta)
+        positive_x_epsilon[:, 0] = epsilon
+        center_radius = torch.linalg.vector_norm(radial_delta + positive_x_epsilon, dim=1) - epsilon
+        return torch.where(at_center, center_radius, radius)
```
[the remark this answers is not in the document]
Comment (lgulich) [reply isaaclab_arena/relations/relation_loss_strategies.py:686]: Addressed in f75befb5. The synthetic center radius now uses clamp_min(0.0), with a regression test that emulates norm under-rounding and verifies a non-negative result.

File: isaaclab_arena/relations/relations.py [hunk 5]
```diff
@@ -493,15 +493,24 @@ def __init__(
         z_min: float | None = None,
         z_max: float | None = None,
         relation_loss_weight: float = 1.0,
+        center_x: float | None = None,
+        center_y: float | None = None,
+        radius_min: float | None = None,
+        radius_max: float | None = None,
     ):
```
Comment (alexmillane) [inline isaaclab_arena/relations/relations.py]: What do you think about separating this into two classes:
- `PositionLimitsBox`
- `PositionLimitsCylindrical`

That way the meaning of these (many) constructor arguments is separated and clear for the user. Right now they're mixed up and a bit difficult to determine what is used where.

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 6]
```diff
@@ -617,5 +620,71 @@ def compute_loss(
...
+                total_loss = total_loss + single_boundary_linear_loss(
+                    self._radius_with_lower_bound_escape_direction(relation, radial_delta, radius),
+                    relation.radius_min,
+                    slope=self.slope,
+                    penalty_side="less",
+                )
+            else:
+                total_loss = total_loss + single_boundary_linear_loss(
+                    radius, relation.radius_max, slope=self.slope, penalty_side="greater"
+                )
```
Comment (alexmillane) [inline isaaclab_arena/relations/relation_loss_strategies.py]: Can we get a comment on each branch here.
- annulus
- keep-out of circle
- stay inside circle.

File: isaaclab_arena/relations/relation_loss_strategies.py (hunk 2 above)
Comment (alexmillane) [inline isaaclab_arena/relations/relation_loss_strategies.py]: Is this a practical scenario? I.e. a scenario that we're likely to run into in situations that are not already degenerate?

Let's talk about this offline.

**Review by alexmillane — COMMENTED** [review]
Couple of comments.

Both of them relate to the combined/separate treatment of box and cylindrical constraints.

Clearly some thinking has gone into how these two interact. I'm wondering if the complexity added by handling this due to realistic issues we're likely to face.

File: isaaclab_arena/relations/relations.py (hunk 5 above)
Comment (lgulich) [reply isaaclab_arena/relations/relations.py]: Done in c379466d8. The public API is now `PositionLimitsBox` and `PositionLimitsCylindrical`, with separate constructor arguments and separate loss strategies. For backward compatibility, Python `PositionLimits` and YAML `position_limits` both alias the box relation.

File: isaaclab_arena/relations/relation_loss_strategies.py (hunk 6 above)
Comment (lgulich) [reply isaaclab_arena/relations/relation_loss_strategies.py]: Added explicit comments for the annulus, keep-out cylinder, and stay-inside cylinder branches in `PositionLimitsCylindricalLossStrategy` in c379466d8.

File: isaaclab_arena/relations/relation_loss_strategies.py (hunk 2 above)
Comment (lgulich) [reply isaaclab_arena/relations/relation_loss_strategies.py]: Agreed. I removed it and box and cylindrical limits now compose as independent relations through the normal summed loss.

File: docs/pages/concepts/concept_object_placement.rst [hunk 7]
```diff
@@ -180,6 +180,67 @@ Parameters:
...
+for the object's bounding-box footprint.
+
+Attach both relations when an object needs the intersection of box and cylindrical
+limits. For example, this keeps an object's origin inside a rectangular tabletop
+region and within 0.25 m of a world-XY point while ``On`` determines its height:
+
+.. code-block:: python
+
+   from isaaclab_arena.relations.relations import (
+       On,
```
Comment (lgulich) [inline docs/pages/concepts/concept_object_placement.rst]: On is not needed

File: docs/pages/references/release_notes.rst [hunk 8]
```diff
@@ -77,7 +77,8 @@ and DROID integration, RL workflows, and a large set of new tasks and embodiment
   final subtask success states, and an example that puts an object into a microwave and
   closes the door (#289, #323, #337, #365).
 - **Semantic object placement:** Added a differentiable relation-based object placement
-  solver supporting ``On``, ``NextTo``, ``AtPosition``, and ``PositionLimits`` relations,
```
Comment (lgulich) [inline docs/pages/references/release_notes.rst:80]: undo this, do not change the existing release notes

File: isaaclab_arena/tests/test_position_limit_class_names.py
Comment (lgulich) [inline isaaclab_arena/tests/test_position_limit_class_names.py:1]: let's remove this file

File: isaaclab_arena/tests/test_position_limits.py
Comment (lgulich) [inline isaaclab_arena/tests/test_position_limits.py:1]: Please double check this file: for the cylindrical constraint only add the equivalent tests taht already existed for the box constraint. do not add conceptually completely new tests.

**Review by lgulich — COMMENTED** [review]
self review

File: docs/pages/concepts/concept_object_placement.rst (hunk 7 above)
Comment (lgulich) [reply docs/pages/concepts/concept_object_placement.rst]: Addressed in 859553117. Removed `On` from the example import and relation attachment, and removed the height-related wording.

File: docs/pages/references/release_notes.rst (hunk 8 above)
Comment (lgulich) [reply docs/pages/references/release_notes.rst:80]: Addressed in 859553117. Restored the release-note file exactly to its pre-PR `PositionLimits` wording.

File: isaaclab_arena/tests/test_position_limit_class_names.py
Comment (lgulich) [reply isaaclab_arena/tests/test_position_limit_class_names.py:1]: Addressed in 859553117. Deleted `test_position_limit_class_names.py`; its standalone API-structure tests were not relocated.

File: isaaclab_arena/tests/test_position_limits.py
Comment (lgulich) [reply isaaclab_arena/tests/test_position_limits.py:1]: Addressed in 859553117. Reduced cylindrical coverage to the historical box counterparts: construction/validation, inside/above/below and single-bound loss behavior, weight scaling, unconstrained Z, and one solver integration test. Removed the cylindrical-only batching, explicit-gradient, underflow, composition, and combined-solver tests.

**Review by alexmillane — APPROVED** [review]
Looks great!
